packages/core/js-peer/src/marine/worker/index.ts

```diff
@@ -49,10 +49,15 @@ export class MarineBackgroundRunner implements IMarineHost {
 
         this.marineServices = new Set();
         await this.workerLoader.start();
+        const worker = await this.workerLoader.getValue();
         await this.controlModuleLoader.start();
-        const worker = this.workerLoader.getValue();
         const wasm = this.controlModuleLoader.getValue();
-        this.workerThread = await spawn<MarineBackgroundInterface>(worker, { timeout: 99999999 });
+        
+        await this.avmWasmLoader.start();
+        
+        console.log('before spawn');
```

Comment:
Don't forget to remove the console.logs that are needed. Or add some logs that you think are important using the logs lib that is used in this repo with logging levels and stuff